packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -1147,23 +1165,35 @@ class _SnappingSimulation extends Simulation {
     if (isDone(time)) {
       return 0;
     }
-    return velocity;
+    if (snapAnimationCurve != null) {
+      // Approximate the derivative using two points.
+      const double delta = 0.001;
+      return (x(time + delta) - x(time)) / delta;
+    } else {
+      return displacement / duration;
+    }
```

Comment:
I was unsure of what to do here. I couldn't see any change in behavior regardless how `dx` was implemented. In the end I went for an approach to approximate the derivative since from what I can see there's no easy way to get the derivative of an arbitrary curve. Let me know what you think or if I should update it.